velox/functions/prestosql/tests/ArrayShuffleTest.cpp

```diff
@@ -0,0 +1,316 @@
+        << "Shuffle " << input->encoding()
+        << " array must yield >= " << kThreshold
+        << " distinct values, but only got " << numDistinctValues;
+  }
+};
+} // namespace
+
+TEST_F(ArrayShuffleTest, normalArrays) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{},
+       {std::nullopt},
+       {std::nullopt, std::nullopt},
+       {-1, 0, 1},
+       {std::nullopt, 0, 0},
+       {std::numeric_limits<int64_t>::max(),
+        std::numeric_limits<int64_t>::min(),
+        0,
+        1,
+        2,
+        3}});
+  testShuffle<int64_t, true>(input);
+}
+
+TEST_F(ArrayShuffleTest, nestedArrays) {
+  using innerArrayType = std::vector<std::optional<int64_t>>;
+  using outerArrayType =
+      std::vector<std::optional<std::vector<std::optional<int64_t>>>>;
+  innerArrayType a{1, 2, 3, 4};
+  innerArrayType b{5, 6};
+  innerArrayType c{6, 7, 8};
+  outerArrayType row1{{a}, {b}};
+  outerArrayType row2{std::nullopt, std::nullopt, {a}, {b}, {c}};
+  outerArrayType row3{{{}}};
+  outerArrayType row4{{{std::nullopt}}};
+  auto input =
+      makeNullableNestedArrayVector<int64_t>({{row1}, {row2}, {row3}, {row4}});
+
+  testShuffle<Array<int64_t>, true>(input);
+}
+
+TEST_F(ArrayShuffleTest, sortAndShuffle) {
+  auto input = makeNullableArrayVector<int64_t>(
+      {{-1, 0, std::nullopt, 1, std::nullopt},
+       {4, 1, 5, 3, 2},
+       {std::numeric_limits<int64_t>::max(),
+        std::numeric_limits<int64_t>::min(),
+        4,
+        1,
+        3,
+        2,
+        std::nullopt}});
+  auto inputVector = makeRowVector({input});
+  auto result1 = evaluate<ArrayVector>("array_sort(C0)", inputVector);
+  auto result2 = evaluate<ArrayVector>("array_sort(shuffle(C0))", inputVector);
+
+  assertEqualVectors(result1, result2);
+}
+
+TEST_F(ArrayShuffleTest, constantEncoding) {
+  vector_size_t size = 100;
+  // Considering the cases of empty array, nullable array, duplicate-element
+  // array, and distinct-value array.
+  auto valueVector = makeNullableArrayVector<int64_t>(
+      {{}, {std::nullopt, 0}, {5, 5}, {1, 2, 3}});
+
+  for (auto i = 0; i < valueVector->size(); i++) {
+    auto input = BaseVector::wrapInConstant(size, i, valueVector);
+    testShuffle<int64_t, true>(input);
+  }
+}
+
+TEST_F(ArrayShuffleTest, dictEncoding) {
+  // Considering the case of repeated elements: {1,2,3} x 3, {4,5} x 2.
+  auto base = makeNullableArrayVector<int64_t>(
+      {{0},
+       {1, 2, 3},
+       {4, 5, std::nullopt},
+       {1, 2, 3},
+       {1, 2, 3},
+       {4, 5, std::nullopt}});
+  // Considering the case of indices filtering (filter out element at index 0)
+  std::vector<vector_size_t> indicesArray({3, 4, 1, 2});
+
+  auto indices = makeIndices(indicesArray);
+  auto input = wrapInDictionary(indices, base);
+
+  testShuffle<int64_t, true>(input);
+}
+
+TEST_F(ArrayShuffleTest, flatEncodingRandomness) {
+  auto inputVector = generateVector(VectorEncoding::Simple::FLAT);
+  testShuffleRandomness(inputVector);
```

Comment:
testShuffleRandomness is always used together with generateVector. Consider folding generateVector logic into testShuffleRandomness and simplify the callers:

```
testShuffleRandomness(VectorEncoding::Simple::FLAT);
```